lib/ecto/adapters/mysql/connection.ex

```diff
@@ -14,6 +14,17 @@ if Code.ensure_loaded?(Mariaex.Connection) do
       Mariaex.Connection.start_link(opts)
     end
 
+    def query(conn, sqls, params, opts) when is_list(sqls) do
+      Enum.reduce sqls, {}, fn(sql, response) ->
```

Comment:
@josevalim I want to make sure you're OK with this. `Mira.Connection.query` only accepts one command at a time. In order to get the MySQL prepared statement working I had to split it up into its separate commands and process each one at a time. This `Enum.reduce` statement will loop through each command. If an `{:error, message}` is encountered by any of the commands it will complete the loop but no longer process any more commands. The error will be returned up the chain as the last command executed.